src/masq.js

```
@@ -288,9 +284,124 @@ class Masq {
+      delete dumpedStore.tokenList
+      delete dumpedStore.deviceList
+      return dumpedStore
+    } catch (error) {
+      if (error.name === 'OperationError') {
+        throw common.generateError(common.ERRORS.WRONGPASSPHRASE)
+      } else {
+        throw (error)
+      }
+    }
+  }
+
+  /**
+   * Import function expects import rules for each application stored in backup.
+   * By default all the applications overwrite rules will be set to true.
+   * The applications list must be shown to user to let him the choice between the
+   * backup app data and the current Masq Store app data.
+   *
+   * @param {string} backup
+   * @returns {object} - The import rules needed by import function
+   */
+  getBackupAppList (backup) {
+    const importRules = {}
+    for (let key of Object.keys(backup.appList)) {
+      importRules[backup.appList[key].url] = true
+    }
+    return importRules
+  }
+
+  /**
+   * Import function, the passphrase is asked to add a layer of protection.
+   * The import rules must follow the following structure :
+   * {
+   *  appName1 : boolean
+   *  appName2 : boolean
+   * }
+   * The boolean corresponds to the overwrite rule, if true the application will be overwritten.
+   *
+   * @param {string} importedStore - The data we want to import
+   * @param {Object} importRules - This object contains the applications in backup and the associated overwrite rules
+   * @param {string} passphrase - The passphrase
+   * @returns {Promise}
+   *
+   */
+  async import (importedStore, importRules, passphrase) {
+    this._checkCurrentUser()
+    if (!importRules || Object.keys(importRules).length === 0) {
+      throw common.generateError(common.ERRORS.NOIMPORTRULES)
+    }
+    const publicProfile = await this.getUser()
+    if (!passphrase || passphrase === '') {
+      throw common.generateError(common.ERRORS.NOPASSPHRASE)
+    }
+
+    const derivedkey = await MasqCrypto.utils.deriveKey(passphrase, publicProfile.salt)
```

Like in a previous comment, can we throw the error from `deriveKey()` ?